offixsdk/src/main/java/org/aerogear/offixsdk/ClientGenerator.kt

```diff
@@ -0,0 +1,139 @@
+                .build()
+        }
+
+        return apClient
+    }
+
+    private fun getOkhttpClient(): OkHttpClient? {
+
+        //Adding HttpLoggingInterceptor() to see the response body and the results.
+        httpClient?.let {
+            return it
+        } ?: kotlin.run {
+            httpClient = OkHttpClient.Builder()
+                .addInterceptor(LoggingInterceptor())
+                .addInterceptor(getResponseInterceptor(applicationContext)!!)
+                .build()
+        }
+        return httpClient
+    }
+
+    //function to get the response after query/mutation is performed, can get conflict messages and so on.
+    private fun getResponseInterceptor(context: Context): Interceptor? {
+
+        conflictInterceptor?.let {
+            return it
+        } ?: kotlin.run {
+            conflictInterceptor = Interceptor {
+                val request = it.request()
+
+                //all the HTTP work happens, producing a response to satisfy the request.
+                val response = it.proceed(request)
+
+                //https://stackoverflow.com/a/33862068/10189663
+
+                val responseBody = response.body()
+                val bufferedSource = responseBody?.source()
+                bufferedSource?.request(Long.MAX_VALUE)
+                val buffer = bufferedSource?.buffer()
+                val responseBodyString = buffer?.clone()?.readString(Charset.forName("UTF-8")) ?: ""
+
+                Log.e("UtillClass", " Interceptor : $responseBodyString")
+
+                //To see for conflict, "VoyagerConflict" which comes in the message is searched for.
+                if (responseBodyString.contains("VoyagerConflict")) {
+                    Log.e(TAG, " Conflcit in offixClient")
+                }
+                return@Interceptor response
+            }
+        }
+        return conflictInterceptor
+    }
+
+    private fun cacheResolver(): CacheKeyResolver {
+        return object : CacheKeyResolver() {
+            override fun fromFieldRecordSet(field: ResponseField, recordSet: Map<String, Any>): CacheKey {
+                Log.e(TAG, "fromFieldRecordSet ${(recordSet["id"] as String)}")
+//                if (recordSet.containsKey("id")) {
+//                    val typeNameAndIDKey = recordSet["__typename"].toString() + "." + recordSet["id"]
+//                    return CacheKey.from(typeNameAndIDKey)
+//                }
+//                return CacheKey.NO_KEY
+                return formatCacheKey(recordSet["id"] as String?)
+            }
+
+            // Use this resolver to customize the key for fields with variables: eg entry(repoFullName: $repoFullName).
+            // This is useful if you want to make query to be able to resolved, even if it has never been run before.
+            override fun fromFieldArguments(field: ResponseField, variables: Operation.Variables): CacheKey {
+                Log.e(TAG, "fromFieldArguments $variables")
```

Comment:
Wrong log  level

Reply:
Why?

Reply:
Dont log the messages as error everywhere 
Use Log.d/v/w for things that aren't errors.